src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -293,7 +296,7 @@ internal class SqlServerFhirDataStore : IFhirDataStore, IProvideCapability
             long baseResourceSurrogateId = ResourceSurrogateIdHelper.LastUpdatedToResourceSurrogateId(resource.LastModified.UtcDateTime);
             short resourceTypeId = _model.GetResourceTypeId(resource.ResourceTypeName);
 
-            if (_schemaInformation.Current >= SchemaVersionConstants.PreventUpdatesFromCreatingVersionWhenNoImpact)
+            if (_schemaInformation.Current >= SchemaVersionConstants.TokenOverflow)
```

Comment:
The reason to differentiate between SchemaVersionConstants.TokenOverflow and SchemaVersionConstants.PreventUpdatesFromCreatingVersionWhenNoImpact is because TokenOverflow adds new columns to the tables. New TokenOverflow case uses autogenerated VLatest.UpsertResourceTvpGenerator which then ends up calling BulkTokenStringCompositeSearchParameterV2RowGenerator. BulkTokenStringCompositeSearchParameterV2RowGenerator translates generic search parameters to a particular storage implementation, in this case SQL db with schema 41 which adds token overflow field. BulkTokenStringCompositeSearchParameterV2RowGenerator is in a new file that is added for this reason.